search/searcher/search_regexp_test.go

```diff
@@ -101,32 +113,21 @@ func testRegexpSearch(t *testing.T, twoDocIndex index.Index,
 	regexpSearcherCo := searcherMaker(t, twoDocIndexReader, "co.*", "desc")
 
 	tests := []struct {
-		searcher  search.Searcher
-		expecteds []*search.DocumentMatch
+		searcher search.Searcher
+		id2score map[string]float64
 	}{
 		{
 			searcher: regexpSearcher,
-			expecteds: []*search.DocumentMatch{
-				{
-					IndexInternalID: internalIDMaker(1),
-					Score:           1.916290731874155,
-				},
-			},
+			id2score: make(map[string]float64),
```

Comment:
You could avoid the adding of entires separately (as in lines 128-130) by initializing the map this way ...
```
id2score: map[string]float64{
    "1": 1.916290731874155,
}
```